src/index.tsx

```diff
@@ -69,14 +69,14 @@ const LayerMaterial = React.forwardRef<
   LAYERS.LayerMaterial,
   React.PropsWithChildren<LayerMaterialProps & Omit<AllMaterialProps, 'color'>>
 >(({ children, ...props }, forwardRef) => {
-  const ref = React.useRef<LAYERS.LayerMaterial>(null!)
+  const ref = React.useRef<LAYERS.LayerMaterial & { __hasRefreshed: boolean }>(null!)
```

Comment:
Add the _needsRefresh or whatever to the LayerMaterial itself? Might be generally useful to have it there, typed and all